### PR TITLE
Add privacy policy/terms and conditions settings and display in bottom of certain pages.

### LINKED ISSUE #12938: Impress and privacy in footer for login-page
Hello Everybody,

du to the new GDPR it is necessary to link the impress and privacy notes on every page, even if it is only a login page. This is the case for the Matomo Backend. Right now there is no possibility to change the template, add a impress or privacy-note to the footer.

Ideal: It should be possible to add custom pages to Matomo to be shown in the footer.

Is there any workaround for this?

Regards

### PR DESCRIPTION
**Pull request by diosmosis**
Login page:

image

Dashboard (for anonymous user):

image

Another normal page (for anonymous user):

image

Widgetized (when enabled):

image

Fixes #12938

### CODE REVIEW

**Comment by diosmosis** [comment]
If just one link is defined:

image

**Comment by tsteur** [comment]
Looks pretty good. I wonder if at some point we also might want to add "Powered by Matomo" there :) Off topic though :)

File: plugins/PrivacyManager/PrivacyManager.php [hunk 1]
```diff
@@ -601,4 +604,36 @@ public static function getUserIdSalt()
...
+        $view->termsAndCondition = $settings->termsAndConditionUrl->getValue();
+        $out .= $view->render();
+    }
+
+    private function shouldRenderFooterLinks(SystemSettings $settings)
+    {
+        $module = Common::getRequestVar('module', false);
+        if ($module == 'Login') {
+            return true;
+        }
```
Comment (sgiehl) [inline plugins/PrivacyManager/PrivacyManager.php]: Actually that check is redundant, as all users on the login pages are `anonymous`. But guess we could leave it there for a better understanding

File: plugins/PrivacyManager/SystemSettings.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * The intellectual and technical concepts contained herein are protected by trade secret or copyright law.
+ * Redistribution of this information or reproduction of this material is strictly forbidden
+ * unless prior written permission is obtained from InnoCraft Ltd.
+ *
+ * You shall use this code only in accordance with the license agreement obtained from InnoCraft Ltd.
+ *
+ * @link https://www.innocraft.com/
+ * @license For license details see https://www.innocraft.com/license
+ */
+namespace Piwik\Plugins\PrivacyManager;
```
Comment (sgiehl) [inline plugins/PrivacyManager/SystemSettings.php]: Can't use that license header here. Maybe we should rebase that branch, so it isn't anywhere in the history

**Review by sgiehl — CHANGES_REQUESTED** [review]

**Comment by diosmosis** [comment]
@sgiehl updated

**Review by mattab — APPROVED** [review]

**Pull request merged**